### PR TITLE
API outside collaborators

### LINKED ISSUE #715: Organization Outside Collaborators
https://developer.github.com/v3/orgs/outside_collaborators/

There is no matching method for this in PyGithub. I need this to make a chatops bot that polls who in an org has 2fa enabled. 

--timball

### PR DESCRIPTION
**Pull request by jappievw**
Fixes #715

### CODE REVIEW

**Comment by PercussiveRepair** [comment]
Works for me. Please merge and release to pip, I really need this! :)

**Comment by joaaogui** [comment]
Yep, me too!

**Comment by sfdye** [comment]
@jappievw Thanks for the PR, LGTM. Would you mind also adding the implementation for [Remove outside collaborator](https://developer.github.com/v3/orgs/outside_collaborators/#remove-outside-collaborator) and [Convert member to outside collaborator](https://developer.github.com/v3/orgs/outside_collaborators/#convert-member-to-outside-collaborator). I will merge in the changes once all the endpoints are there for org outside collaborators.

**Comment by jappievw** [comment]
Check, I can extend it with those as soon as I'm back from my holidays!

**Comment by sfdye** [comment]
That’s awesome, enjoy the vacation @jappievw

**Comment by jappievw** [comment]
@Sieabah Yeap, soonish, actually just rebased my fork!

**Comment by csidell-playq** [comment]
Awesome!

**Comment by Sieabah** [comment]
Awesome!

**Comment by jappievw** [comment]
Just implemented the two remaining API calls @sfdye. Sorry for the delay, stretched the holiday a bit😉.

File: github/Organization.py [hunk 1]
```diff
@@ -510,6 +510,49 @@ def get_public_members(self):
...
+        return github.PaginatedList.PaginatedList(
+            github.NamedUser.NamedUser,
+            self._requester,
+            self.url + "/outside_collaborators",
+            url_parameters
+        )
+
+    def remove_outside_collaborator(self, collaborator):
+        """
+        :calls: `DELETE /orgs/:org/outside_collaborators/:username <http://developer.github.com/v3/orgs/members>`_
```
Comment (sfdye) [inline github/Organization.py]: Maybe this url should point to?
https://developer.github.com/v3/orgs/outside_collaborators

File: github/Organization.py [hunk 2]
```diff
@@ -510,6 +510,49 @@ def get_public_members(self):
...
+        """
+        assert isinstance(collaborator, github.NamedUser.NamedUser), collaborator
+        headers, data = self._requester.requestJsonAndCheck(
+            "DELETE",
+            self.url + "/outside_collaborators/" + collaborator._identity
+        )
+
+    def convert_to_outside_collaborator(self, member):
+        """
+        :calls: `PUT /orgs/:org/outside_collaborators/:username <http://developer.github.com/v3/orgs/members>`_
```
Comment (sfdye) [inline github/Organization.py]: Here too

File: github/tests/Organization.py [hunk 3]
```diff
@@ -108,6 +108,15 @@ def testGetIssuesWithAllArguments(self):
...
         self.assertListKeyEqual(self.org.get_members(), lambda u: u.login, ["cjuniet", "jacquev6", "Lyloa"])
 
+    def testGetOutsideCollaborators(self):
+        self.assertListKeyEqual(self.org.get_outside_collaborators(), lambda u: u.login, ["octocat"])
+
+    def testOutsideCollaborators(self):
+        octocat = self.g.get_user("octocat")
+        self.org.convert_to_outside_collaborator(octocat)
+        self.assertListKeyEqual(self.org.get_outside_collaborators(), lambda u: u.login, ["octocat"])
+        self.org.remove_outside_collaborator(octocat)
```
Comment (sfdye) [inline github/tests/Organization.py]: Maybe one more assert after the delete?

**Comment by sfdye** [comment]
@jappievw Left some comments, just minor change

File: github/tests/Organization.py (hunk 3 above)
Comment (jappievw) [reply github/tests/Organization.py]: Done!

**Comment by jappievw** [comment]
@sfdye just processed your feedback and amended the changes!

**Comment by sfdye** [comment]
@jappievw Awesome, thanks!

**Comment by sfdye** [comment]
@csidell-playq Soon, once we close a few remaining issues.

**Comment by sfdye** [comment]
@csidell-playq just released in `1.40`

**Pull request merged**